### ISSUE TITLE
"x" operation far too cpu-hungry

### ISSUE THREAD

**Issue by jleclanche** [VSCodeVim/Vim#1581]
* Click *thumbs-up* 👍 on this issue if you want it!
* Click *confused* 😕 on this issue if not having it makes VSCodeVim unusable. 

The VSCodeVim team prioritizes issues based on reaction count. 

--------

### What did you do?

- Create a document with lines of varied length
- Go to the middle of a line with 50 characters or so, type `1000x`
- Operation takes several seconds.
- Do the same with `10000x`, `100000x`. Or don't, you might be there a while.

* VSCode Version: 1.11.2 @ 6eaebe3b9c70406d67c97779468c324a7a95db0e
* VsCodeVim Version: 0.6.20
* OS: Linux

**Comment by johnfn** [comment]
As a workaround, try D.

**Comment by jleclanche** [comment]
Yeah, no problem. I'm used to using x. It's still concerning it can hang the editor pretty bad tho :)

**Comment by llarsson** [comment]
What does the plugin do when I hit `x` in Normal mode? Is it identically the same as when I hit my Delete key while in Insert mode? And is that identically the same as when I just do it in "vanilla" VS Code? Because if so, I think the problem is more on VS Code's handling than on this plugin.

**Comment by johnfn** [comment]
Yes, it's equivalent to a delete press. However, that doesn't absolve VSCode! 10000 single delete presses of a single character are about 10,000x slower than a single delete press on a 10,000 character range.

**Comment by llarsson** [comment]
Sure, but selecting 10000 characters and deleting them is not the same as hitting Delete that many times, as only the latter will e.g. remove matching parentheses.

Grant Mathews <notifications@github.com> skrev: (27 april 2017 11:53:19 CEST)
>Yes, it's equivalent to a delete press. However, that doesn't absolve
>VSCode! 10000 single delete presses of a single character are about
>10,000x slower than a single delete press on a 10,000 character range. 
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/VSCodeVim/Vim/issues/1581#issuecomment-297668654

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by jleclanche** [comment]
But `1000x` should not be "hit delete 1000 times". Right now, it inserts 1000 items in the undo stack for example (thankfully only one in the vim undo, but ctrl+z gets 1000 of them).

**Comment by rebornix** [comment]
We may want to improve some batch operations instead of running them in sequence if I understand this problem correctly. Fair request.

**Comment by johnfn** [comment]
What I was trying to say, unclearly like normal, is that if 1000x would be handled as a single delete on a range from start to start + 1000, this would never be an issue.

**Comment by bluesd** [comment]
Dunno if related, think so, but if you hold `x` or `del` for a moment, it takes a lot of time to "end the command", I mean it keeps deleting chars way after you release the x/del key.
